lib/kdump_utils.pm

```diff
@@ -164,4 +169,107 @@ sub do_kdump {
     script_run "echo c > /proc/sysrq-trigger", 0;
 }
 
+sub install_service {
+    zypper_call("in kdump");
+}
+
+sub enable_service {
+    activate_kdump();
+}
+
+sub start_service {
+    systemctl 'start kdump';
+}
+
+# check service is running and enabled
+sub check_service {
+    systemctl 'is-enabled kdump.service';
+    systemctl 'is-active kdump';
+}
+
+sub configure_service {
+    my ($test_type) = @_;
+    $test_type //= '';
+
+    my $self = y2_module_consoletest->new();
+    # preparation for crash test
+    if (is_sle '15+' and $test_type eq 'function') {
+        add_suseconnect_product('sle-module-desktop-applications');
+        add_suseconnect_product('sle-module-development-tools');
+    }
+    if ($test_type eq 'function') {
+        prepare_for_kdump;
+    }
+
+    activate_kdump;
+
+    # restart to activate kdump
+    power_action('reboot', keepconsole => is_spvm);
+    reconnect_mgmt_console if is_spvm;
+    $self->wait_boot;
+
+    select_console 'root-console';
+    if (check_var('ARCH', 'ppc64le') || check_var('ARCH', 'ppc64')) {
+        if (script_run('kver=$(uname -r); kconfig="/boot/config-$kver"; [ -f $kconfig ] && grep ^CONFIG_RELOCATABLE $kconfig')) {
+            record_soft_failure 'poo#49466 -- No kdump if no CONFIG_RELOCATABLE in kernel config';
+            return 1;
+        }
+    }
+    check_service();
+}
+
+sub check_function {
+    my ($test_type) = @_;
+    $test_type //= '';
+
+    my $self = y2_module_consoletest->new();
+
+    # often kdump could not be enabled: bsc#1022064
+    return 1 unless kdump_is_active;
+
+    do_kdump;
+
+    if (get_var('FADUMP')) {
+        reconnect_mgmt_console;
+        assert_screen 'grub2', 180;
+        wait_screen_change { send_key 'ret' };
+    }
+    elsif (is_spvm) {
+        reconnect_mgmt_console;
+    }
+    else {
+        power_action('reboot', observe => 1, keepconsole => 1);
+    }
+
+    # Wait for system's reboot; more time for Hyper-V as it's slow.
+    $self->wait_boot(bootloader_time => check_var('VIRSH_VMM_FAMILY', 'hyperv') ? 200 : undef);
+    select_console 'root-console';
+
+    # all but PPC64LE arch's vmlinux images are gzipped
+    my $suffix = check_var('ARCH', 'ppc64le') ? '' : '.gz';
+    assert_script_run 'find /var/crash/';
+
+    if ($test_type eq 'function') {
+        my $crash_cmd = "echo exit | crash `ls -1t /var/crash/*/vmcore | head -n1` /boot/vmlinux-`uname -r`$suffix";
+        validate_script_output "$crash_cmd", sub { m/PANIC:\s([^\s]+)/ }, 600;
+    }
+    assert_script_run 'rm -fr /var/crash/*';
+}
+
+# check kdump service before and after migration
+# stage is 'before' or 'after' system migration.
+sub full_kdump_check {
+    my ($stage) = @_;
+    $stage //= '';
+
+    select_console 'root-console';
+
+    if ($stage eq 'before') {
+        install_service();
+        configure_service('before');
```

Comment:
also not exist enable service?

Reply:
we use activate_kdump to enable kdump service.